californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -611,11 +610,22 @@ public InetSocketAddress getAddress() {
 	public URI getUri() {
 		URI uri = null;
 		try {
-			InetSocketAddress address = getAddress();
+			InetSocketAddress socketAddress = getAddress();
 			String scheme = CoAP.getSchemeForProtocol(connector.getProtocol());
-			uri = new URI(scheme, null, address.getHostString(), address.getPort(), null, null, null);
-		} catch (URISyntaxException e) {
-			LOGGER.warn("URI", e);
+			String host = socketAddress.getAddress().getHostAddress();
```

Comment:
Is it normal that `socketAddress.getAddress().getHostAddress()` are using instead of `address.getHostString()`

Reply:
For a local networkinterface the hostname seems for me to be not that important.

Reply:
It makes sense.

So about #567, the last address.getHostString() which could cause issue is in `TLSClientConnector.createSllEngine`/`TLSServerConnector.createSllEngine`